common.h still use function typeof,
this was never defined part of c++ and is
not supported in c++11 anymore
so we revert back to gnu++03 as it is not fixed by upstream
Signed-of-by: Joerg Bornkessel <hd_brummy@gentoo.org> (2016/aug/01)
diff -Naur satip-2.2.3.orig/Makefile satip-2.2.3/Makefile
--- satip-2.2.3.orig/Makefile	2016-08-03 23:17:36.769080128 +0200
+++ satip-2.2.3/Makefile	2016-08-03 23:18:36.399080128 +0200
@@ -32,6 +32,7 @@
 
 export CFLAGS   = $(call PKGCFG,cflags)
 export CXXFLAGS = $(call PKGCFG,cxxflags)
+export CXXFLAGS += -std=gnu++03
 STRIP = true
 
 ### The version number of VDR's plugin API:
diff -Naur satip-2.2.3.orig/common.h satip-2.2.3/common.h
--- satip-2.2.3.orig/common.h	2016-08-07 02:54:27.290440336 +0200
+++ satip-2.2.3/common.h	2016-08-07 02:54:53.193773669 +0200
@@ -52,7 +52,7 @@
   do {                                                       \
      if (exp) {                                              \
         char tmp[64];                                        \
-        esyslog("[%s,%d]: "errstr": %s", __FILE__, __LINE__, \
+        esyslog("[%s,%d]: " errstr ": %s", __FILE__, __LINE__, \
                 strerror_r(errno, tmp, sizeof(tmp)));        \
         func;                                                \
         ret;                                                 \
